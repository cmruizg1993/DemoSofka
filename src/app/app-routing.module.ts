import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';
import { ProductRoutingModule } from './product/product-routing.module';

const routes: Routes = [];

@NgModule({
  imports: [
    RouterModule.forRoot(routes), ProductRoutingModule
  ],
  exports: [RouterModule]
})
export class AppRoutingModule { }
