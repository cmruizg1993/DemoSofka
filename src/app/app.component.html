
<app-loader>
    
</app-loader>
<div [class.d-none]="loading"   class="container-fluid bg-secondary">
    <app-header></app-header>
    <div>
        <router-outlet></router-outlet>
    </div>
</div>

