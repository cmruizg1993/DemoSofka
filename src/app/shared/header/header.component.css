.header{
    align-items: center;
    padding: 1rem;
    box-shadow: 1px 10px 4px 2px rgba(213,209,209,0.69);
    -webkit-box-shadow: 1px 10px 4px 2px rgba(213,209,209,0.69);
    -moz-box-shadow: 1px 10px 4px 2px rgba(213,209,209,0.69);
}
.header-content{
    text-align: center;
    vertical-align: middle;
}
.header-icon{
    width: 2rem;
}
.header-title{
    margin-left: 1rem;
    letter-spacing: 0.5px;
    font-size: 2rem;
    font-weight: bolder;
}