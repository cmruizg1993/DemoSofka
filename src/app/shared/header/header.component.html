<div class="header bg-white mb-2">
    <div class="header-content">
        <img src="assets/img/icon.png" alt=""  class="header-icon">
        <span class="header-title text-primary">BANCO</span>
    </div>
</div>