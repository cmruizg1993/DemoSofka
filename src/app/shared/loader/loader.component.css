.loader-container{
    display: flex;
    height: 100vh;
  }
  .loader {
    margin: auto;
    align-self: center;
    width: 15px;
    aspect-ratio: 1;
    border-radius: 50%;
    animation: l5 1s infinite linear alternate;
  }
  