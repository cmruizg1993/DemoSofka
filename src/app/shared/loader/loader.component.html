<div [class.d-none]="!loading" class="loader-container">
    <div class="loader"></div>
</div>