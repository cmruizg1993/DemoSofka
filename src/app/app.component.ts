import { Component } from '@angular/core';
import { LoaderService } from './services/loader.service';
import { Subscription } from 'rxjs';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrl: './app.component.css'
})
export class AppComponent {
  title = 'CristianRuiz';
  subscription = new Subscription();
  loading = true;
  constructor(private loaderService: LoaderService){
    
  }
  ngOnInit(): void {
    //Called after the constructor, initializing input properties, and the first call to ngOnChanges.
    //Add 'implements OnInit' to the class.
    this.subscription = this.loaderService.loaderStatus.subscribe(state => this.loading = state);
    setTimeout(()=>{
      this.loaderService.currentStatus();
    }, 0)
  }

}
