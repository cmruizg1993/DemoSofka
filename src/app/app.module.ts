import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';
import { HeaderComponent } from './shared/header/header.component';
import { ProductModule } from './product/product.module';
import { LoaderComponent } from './shared/loader/loader.component';
import { HTTP_INTERCEPTORS } from '@angular/common/http';
import { LoadingInterceptor } from './interceptors/loading-interceptor.interceptor';

@NgModule({
  declarations: [
    AppComponent,
    HeaderComponent,
    LoaderComponent,
    
  ],
  imports: [
    BrowserModule,
    AppRoutingModule,
    ProductModule
  ],
  providers: [
    {
      provide: HTTP_INTERCEPTORS,
      useClass: LoadingInterceptor,
      multi: true
    }
  ],
  bootstrap: [AppComponent]
})
export class AppModule { }
