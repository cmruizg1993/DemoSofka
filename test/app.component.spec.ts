import { ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { AppComponent } from '../src/app/app.component';
import { LoaderComponent } from '../src/app/shared/loader/loader.component';
import { HeaderComponent } from '../src/app/shared/header/header.component';


describe('AppComponent', () => {
  let fixture: ComponentFixture<AppComponent>;
  let compiled: HTMLElement;
  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [
        RouterTestingModule
      ],
      declarations: [
        AppComponent, LoaderComponent, HeaderComponent
      ],
    }).compileComponents();
    fixture = TestBed.createComponent(AppComponent);
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  it('Se debe crear la app', () => {
    const app = fixture.componentInstance;
    expect(app).toBeTruthy();
  });

  it(`Debe coincidir con el snapshot'`, () => {
    const app = fixture.componentInstance;
    expect(compiled).toMatchSnapshot();
  });


});
