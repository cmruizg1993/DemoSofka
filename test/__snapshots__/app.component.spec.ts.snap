// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent Debe coincidir con el snapshot' 1`] = `
<div
  id="root1"
>
  <app-loader>
    <div
      class="loader-container d-none"
    >
      <div
        class="loader"
      />
    </div>
  </app-loader>
  <div
    class="container-fluid bg-secondary d-none"
  >
    <app-header>
      <div
        class="header bg-white mb-2"
      >
        <div
          class="header-content"
        >
          <img
            alt=""
            class="header-icon"
            src="assets/img/icon.png"
          />
          <span
            class="header-title text-primary"
          >
            BANCO
          </span>
        </div>
      </div>
    </app-header>
    <div>
      <router-outlet />
      
    </div>
  </div>
</div>
`;
